pkg/flow/adapter/dataweavetransformation/adapter.go

```diff
@@ -96,49 +115,97 @@ func (a *dataweaveTransformAdapter) Start(ctx context.Context) error {
 
 func (a *dataweaveTransformAdapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
 	var err error
-	var tmpfile *os.File
-	path := "/tmp/dw"
-	dwFolder := path + "/custom"
+	var tmpFile *os.File
+	var inputData []byte
+	var spell string
+	var inputContentType string
+	var outputContentType string
 
-	switch a.incomingContentType {
-	case "application/json":
-		if !isValidJSON(event.Data()) {
-			return a.replier.Error(&event, targetce.ErrorCodeRequestParsing,
-				errors.New("invalid Json"), nil)
+	err = validateContentType(event.DataContentType(), event.Data())
+	if err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New(err.Error()), nil)
+	}
+
+	req := &DataWeaveTransformationStructuredRequest{}
+	if err := event.DataAs(req); err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestParsing, err, nil)
+	}
+
+	if !a.spellOverride && req.Spell != "" {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New("it is not allowed to override Spell per CloudEvent"), nil)
+	}
+
+	if a.defaultSpell == nil && req.Spell == "" {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New("no default Spell or in request found"), nil)
+	}
+
+	if req.InputContentType != "" || req.OutputContentType != "" || req.Spell != "" {
+		if req.InputData == "" {
+			return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+				errors.New("inputData not found"), nil)
 		}
-		tmpfile, err = ioutil.TempFile(path, "*.json")
+	}
+
+	// Check for spellOverride to be enabled and all the parameters to be present.
+	if a.spellOverride && req.InputData != "" && req.InputContentType != "" && req.OutputContentType != "" && req.Spell != "" {
+		inputData = []byte(req.InputData)
+		spell = req.Spell
+		inputContentType = req.InputContentType
+		outputContentType = req.OutputContentType
+
+		// In the case that it receives the inputData in a dict instead of directly,
+		// it will use the default parameters present in the yaml and the inputData.
+	} else {
+		if req.InputData != "" {
+			inputData = []byte(req.InputData)
+		} else {
+			inputData = event.Data()
+		}
+		if a.defaultSpell != nil && a.defaultInputContentType != nil && a.defaultOutputContentType != nil {
+			spell = *a.defaultSpell
+			inputContentType = *a.defaultInputContentType
+			outputContentType = *a.defaultOutputContentType
+		}
+	}
+
+	err = validateContentType(inputContentType, inputData)
+	if err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New(err.Error()), nil)
+	}
+
+	switch inputContentType {
+	case "application/json":
+		tmpFile, err = ioutil.TempFile(path, "*.json")
```

Comment:
The file extension doesn't really matter when invoking `dw` given you are explicitly specifying the input/output content-types.